src/Wallabag/CoreBundle/Helper/EntriesExport.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Wallabag\CoreBundle\Helper;
+
+use PHPePub\Core\EPub;
+use PHPePub\Core\Structure\OPF\DublinCore;
+
+class EntriesExport
+{
+    private $format;
+    private $method;
+    private $title;
+    private $entries;
+    private $authors = array('wallabag');
+    private $language;
+    private $tags;
+
+    public function __construct($entries)
+    {
+        $this->entries = $entries;
+
+        foreach ($entries as $entry) {
+            $this->tags[] = $entry->getTags();
+        }
+        if (count($entries) === 1) {
+            $this->language = $entries[0]->getLanguage();
+        }
+    }
+
+    /**
+     * Sets the category of which we want to get articles, or just one entry.
+     *
+     * @param string $method Method to get articles
+     */
+    public function setMethod($method)
+    {
+        $this->method = $method;
+
+        switch ($this->method) {
+            case 'All':
+                $this->title = 'All Articles';
+                break;
+            case 'Unread':
+                $this->title = 'Unread articles';
+                break;
+            case 'Starred':
+                $this->title = 'Starred articles';
+                break;
+            case 'Archive':
+                $this->title = 'Archived articles';
+                break;
+            case 'entry':
+                $this->title = $this->entries[0]->getTitle();
+                break;
+            default:
+                break;
```

Comment:
Since won't affect this part.
Category from the route will only limit the categories you can get. If someone provide a wrong category in the route, the Router won't match the route and will lead to a 404.

You'll still have the `$method` here.

What I said above is that you don't need the switch / case,  just a simple:

```php
$method = ucfirst($category);
```